src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,16 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] args) {
+        Machine excavator = new Excavator();
+        Machine truck = new Truck();
+        Bulldozer bulldozer = new Bulldozer();
```

Comment:
You should use an abstract reference for the `bulldozer` variable as well. According to the checklist: 'Use abstract references instead of specific ones where possible.' Change `Bulldozer bulldozer = new Bulldozer();` to `Machine bulldozer = new Bulldozer();`.